drivers/SmartThings/zigbee-button/src/zigbee-multi-button/heiman/init.lua

```diff
@@ -56,13 +59,25 @@ local do_configure = function(self, device)
   end
 end
 
+local group_button_mapping = {
+  [0x0010] = "button1",
+  [0x0011] = "button3",
+  [0x0012] = "button2",
+}
+
 local function scenes_cluster_handler(driver, device, zb_rx)
   local additional_fields = {
     state_change = true
   }
-  local bytes = zb_rx.body.zcl_body.body_bytes
-  local button_num = bytes:byte(3)
-  local button_name = "button" .. button_num
+
+  local button_name = nil
+  if device:get_model() == "HS6SSB-W-EF-3.0" then
+    button_name = group_button_mapping[zb_rx.address_header.dest_addr.value]
+  else
+    local bytes = zb_rx.body.zcl_body.body_bytes
```

Comment:
Hi, @greens.
I'm a newbie at EdgaDriver.
I can't get the zb_rx.body.zcl_body.body_bytes as well as SceneSwitch-EM-3.0.
So I use the group_button_mapping to get the button_name.

I'm curious about it because in DTH there is no difference between SceneSwitch-EM-3.0 and 
HS6SSB-W-EF-3.0 in sending events to components...

Reply:
@liuyd96 You don't have any trouble, right? you just want to confirm about the implementation method.
I think there is no issue.

Reply:
Hi, @greens 
Please ignore this noise.  I made a mistake...

There are 2 devices: [SceneSwitch-EM-3.0](https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/eb3cee1775bc55148813909aa9e891631de1e2e8/devicetypes/smartthings/zigbee-multi-button.src/zigbee-multi-button.groovy#L33) and [E-SceneSwitch-EM-3.0](https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/eb3cee1775bc55148813909aa9e891631de1e2e8/devicetypes/smartthings/zigbee-scene-keypad.src/zigbee-scene-keypad.groovy#L32C151-L32C171)

Current EdgeDriver supports SceneSwitch-EM-3.0 which is different from HS6SSB-W-EF-3.0...

Sorry for the confusion...

Reply:
I'm just wondering about this code because as far as I can tell there's nothing equivalent in the DTH.

Reply:
> I'm just wondering about this code because as far as I can tell there's nothing equivalent in the DTH.

Hi, @greens 

Here is the log when I press button of HS6SSB-W-EF-3.0
Button1
```
2023-08-09T05:19:49.937119880+00:00 INFO Zigbee Button  <ZigbeeDevice: 7bc48995-7e50-4f91-b9b8-c41387336520 [0x5B28] (HEIMAN Scene Panel)> received Zigbee message: < ZigbeeMessageRx || type: 0x01, < AddressHeader || src_addr: 0x5B28, src_endpoint: 0x01, dest_addr: 0x0010, dest_endpoint: 0xFF, profile: 0x0104, cluster: Scenes >, lqi: 0xB2, rssi: -64, body_length: 0x0008, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x01, seqno: 0x17, ZCLCommandId: 0x05 >, < RecallScene || group_id: 0x0010, scene_id: 0x02, transition_time: 0x0001 > > >
```

Button 2
```
2023-08-09T05:19:51.587393664+00:00 INFO Zigbee Button  <ZigbeeDevice: 7bc48995-7e50-4f91-b9b8-c41387336520 [0x5B28] (HEIMAN Scene Panel)> received Zigbee message: < ZigbeeMessageRx || type: 0x01, < AddressHeader || src_addr: 0x5B28, src_endpoint: 0x01, dest_addr: 0x0012, dest_endpoint: 0xFF, profile: 0x0104, cluster: Scenes >, lqi: 0xA4, rssi: -63, body_length: 0x0008, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x01, seqno: 0x18, ZCLCommandId: 0x05 >, < RecallScene || group_id: 0x0012, scene_id: 0x04, transition_time: 0x0001 > > >
```

Button 3
```
2023-08-09T05:19:53.594695179+00:00 INFO Zigbee Button  <ZigbeeDevice: 7bc48995-7e50-4f91-b9b8-c41387336520 [0x5B28] (HEIMAN Scene Panel)> received Zigbee message: < ZigbeeMessageRx || type: 0x01, < AddressHeader || src_addr: 0x5B28, src_endpoint: 0x01, dest_addr: 0x0011, dest_endpoint: 0xFF, profile: 0x0104, cluster: Scenes >, lqi: 0xC0, rssi: -66, body_length: 0x0008, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x01, seqno: 0x19, ZCLCommandId: 0x05 >, < RecallScene || group_id: 0x0011, scene_id: 0x03, transition_time: 0x0001 > > >
```

Button1 dest_addr is 0x0010
Button2 dest_addr is 0x0012
Button3 dest_addr is 0x0011

In order to send the event to the corresponding endpoint. I added these codes.
Match the dest_addr and button_name
```
  if device:get_model() == "HS6SSB-W-EF-3.0" then
    button_name = group_button_mapping[zb_rx.address_header.dest_addr.value]
  else
    local bytes = zb_rx.body.zcl_body.body_bytes
```

Is it elevant to [zigbee-scene-keypad DTH code] ? (https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/eb3cee1775bc55148813909aa9e891631de1e2e8/devicetypes/smartthings/zigbee-scene-keypad.src/zigbee-scene-keypad.groovy#L197)

Thanks.

Reply:
I think the `data[2]` from the DTH corresponds to the `scene_id` field. We didn't parse fields in the Groovy implementation.

That would line up with this map from the DTH:
```
  "HS6SSB-W-EF-3.0" : [
    "02" : 1,
    "03" : 3,
    "04" : 2
  ]
```

Reply:
> I think the `data[2]` from the DTH corresponds to the `scene_id` field. We didn't parse fields in the Groovy implementation.
> 
> That would line up with this map from the DTH:
> 
> ```
>   "HS6SSB-W-EF-3.0" : [
>     "02" : 1,
>     "03" : 3,
>     "04" : 2
>   ]
> ```

Thanks for your explanation. 
So I need to use ```scene_id``` to map button instead of dest_addr.

Reply:
Hi, @greens 
I've updated the codes to use ```scene_id``` to map button, same with DTH code.

Please check. Thanks.

Reply:
That's okay though, because that code is meant for SceneSwitch-EM-3.0, right? Or am I misunderstanding?

Since this is a significant change in functionality for this driver, could you write some unit tests for this device?

Reply:
> That's okay though, because that code is meant for SceneSwitch-EM-3.0, right? Or am I misunderstanding?
Yes. You're right.


> Since this is a significant change in functionality for this driver, could you write some unit tests for this device?

Done.  

Thanks for your review : )